File: src_java/ch/konnexions/orabench/actions/Select.java
/*
 * 
 */

package ch.konnexions.orabench.actions;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;

import ch.konnexions.orabench.utils.Config;
import ch.konnexions.orabench.utils.Logger;

public class Select implements Runnable {
    private ArrayList<String[]> bulkDataPartition;

    private Config config;

    public Logger log;

    private int partitionKey;

    private Statement statement;

    /**
     * Instantiates a new Select class.
     *
     * @param config            the configuration parameters
     * @param log               the logger
     * @param statement         the statement
     * @param bulkDataPartition the bulk data partition
     * @param partitionKey      the partition key
     */
    public Select(Config config, Logger log, Statement statement, ArrayList<String[]> bulkDataPartition, int partitionKey) {
        this.config = config;
        this.log = log;
        this.statement = statement;
        this.bulkDataPartition = bulkDataPartition;
        this.partitionKey = partitionKey;
    }

    /**
     * Run.
     */
    @Override
    public void run() {
        int count = 0;

        try {
            if (config.getConnectionFetchSize() > 0) {
                statement.setFetchSize(config.getConnectionFetchSize());
            }

            ResultSet resultSet = statement.executeQuery(config.getSqlSelect() + " WHERE partition_key = " + Integer.toString(partitionKey));

            while (resultSet.next()) {
                count += 1;
            }

            if (count != bulkDataPartition.size()) {
                log.error("Number rows: expected=" + Integer.toString(bulkDataPartition.size()) + " - found=" + Integer.toString(count));
            }

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

}

File: src_kotlin/src/main/kotlin/ch/konnexions/Select.kt
package main.kotlin.ch.konnexions

import org.apache.logging.log4j.Logger
import java.sql.Statement

class Select(
    logger: Logger,
    isDebug: Boolean, statement: Statement, bulkDataPartition: ArrayList<Array<String>>, partitionKey: Int, connectionFetchSize: Int, sqlSelect: String
) : Runnable {
    private val bulkDataPartition: ArrayList<Array<String>>
    private val connectionFetchSize: Int
    private val isDebug: Boolean
    private val logger: Logger
    private val partitionKey: Int
    private val sqlSelect: String
    private val statement: Statement

    /**
     * Runs the thread implementer.
     */
    override fun run() {
        if (isDebug) {
            logger.debug("Start")
        }

        OraBench.selectHelper(
            logger,
            isDebug,
            statement,
            bulkDataPartition,
            partitionKey,
            connectionFetchSize,
            sqlSelect
        )

        if (isDebug) {
            logger.debug("End")
        }
    }

    /**
     * Instantiates a new Select class.
     */
    init {
        if (isDebug) {
            logger.debug("Start")
        }

        this.bulkDataPartition = bulkDataPartition
        this.connectionFetchSize = connectionFetchSize
        this.isDebug = isDebug
        this.logger = logger
        this.partitionKey = partitionKey
        this.sqlSelect = sqlSelect
        this.statement = statement

        if (isDebug) {
            logger.debug("End")
        }
    }
}
